engine/gamesys/src/gamesys/components/comp_collection_factory.cpp

```diff
@@ -103,11 +123,14 @@ namespace dmGameSystem
 
     dmGameObject::CreateResult CompCollectionFactoryDestroy(const dmGameObject::ComponentDestroyParams& params)
     {
-        FactoryWorld* fw = (FactoryWorld*)params.m_World;
+        CollectionFactoryWorld* fw = (CollectionFactoryWorld*)params.m_World;
         CollectionFactoryComponent* fc = (CollectionFactoryComponent*)*params.m_UserData;
         CleanupAsyncLoading(dmScript::GetLuaState(((CollectionFactoryContext*)params.m_Context)->m_ScriptContext), fc);
         uint32_t index = fc - &fw->m_Components[0];
         fc->m_Resource = 0x0;
+        if (fc->m_CustomResource)
+            dmGameSystem::ResCollectionFactoryDestroyResource(fw->m_Factory, fc->m_CustomResource);
```

Comment:
If we are still holding this resource, we need to delete it and dereference it's dependencies.